core/helpers/email_helpers.py

```diff
@@ -71,5 +78,29 @@ def report_notify_email(report, email_template_location, **context):
         send_email.render_template_fields(
             context=params, jinja_env=context["dag"].get_template_env()
         )
-        logging.info(f'Sending "{send_email.subject}" email...')
-        send_email.execute(context)
+
+        if use_sendgrid:
```

Comment:
I think this is the wrong place to branch operations.  Regardless of the email system, `send_email.execute` should be the command.  I think you need to change how the EmailOperator is instantiated.